en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
+- Do not suffix classes if they are a subclass any of the following:
+  - `DataObject`
+- Suffix interfaces with `Interface`
+- Suffix traits with `Trait`
+- Prefix abstract classes with `Abstract`
+
+### Other naming conventions
+
+- Extensions should match the name of the class they are extending. For example, an extension class that's added to LeftAndMain should be called LeftAndMainExtension
+- When choosing names for variables and methods use the guiding principle "make it as obvious as possible"
+- Ensure that variable names and method naming is consistent. For example, if you are creating a new class and have a variable named `$itemValue`` which is passed to another method in the same class, the paramater name for that method should also be `$itemValue``.
+
+## Directory naming - src dir
+
+- Put similar types of classes in the same directory. For example, put Controllers in a Controllers directory
+- Put DataObjects in a directory called Models
+- Do not group different class types together in a directory relating to a single feature
+- Use initial caps pluralised for the directory naming e.g. use `Controllers` rather than `Controller` or `controllers`
+
+## Properties and accessors
+
+- Do not use public properites, instead use a combination of private properites and public getters and setters
+- For setters use a combination of a `:static` return type and `return $this` to allow for fluent programming
+- Do not use [dynamic properties](https://www.php.net/manual/en/language.oop5.properties.php#language.oop5.properties.dynamic-properties)
+
+## Method and property visibility
+
+- Use private over protected as this reduces the maintained API surface. Only use protected if there's an immediate reason to upgrade from private.
+- Note that there is a lot of legacy Silverstripe code where protected was used over private.
+
+## Interfaces
+
+- If there are or there's a realistic expection that there will be multiple class implementations of something then create an interface and use that for parameter and return types, and as the service key for injector.
+- If there is only a single implemention and no forseeable expectation of more implementaitons being added, then do not create an interface for the single class implementation because this would only be adding unecessarily complexity.
+
+## Typing
+
+- Aim to strongly type all new properties, method parameters, and method return types.
+- The exception to this is adding a new method or property to a class which is entirely untyped and adding typed API would be very inconsistent with the rest of the file. In this instance use PHP docblocks to define types.
```

Comment:
I think the general approach here was to still add the types to new methods/props so that it's added gradually wherever we touch something, rather than not adding it and creating more technical debt that will need to be addressed at some point anyway.

Reply:
I agree with Michal

```suggestion
```

Reply:
OK work for me

Reply:
Updated